lib/fs.js

```diff
@@ -2014,30 +2014,49 @@ function ReadStream(path, options) {
   this.closed = false;
 
   if (this.start !== undefined) {
-    if (typeof this.start !== 'number' || Number.isNaN(this.start)) {
+    if (typeof this.start !== 'number') {
       throw new ERR_INVALID_ARG_TYPE('start', 'number', this.start);
     }
-    if (this.end === undefined) {
-      this.end = Infinity;
-    } else if (typeof this.end !== 'number' || Number.isNaN(this.end)) {
-      throw new ERR_INVALID_ARG_TYPE('end', 'number', this.end);
+
+    if (Number.isNaN(this.start)) {
```

Comment:
I would rather throw a little more descriptively, let the user know what went wrong.

That said, I'd love to hear everyone's opinions on this, and would love to change this accordingly.

Reply:
There are a couple PRs that consolidate validation functions. It is possible to improve the check here performance and code wise but functionality wise it is fine and I believe it is best to land it as is. It can still be improved later on.

Reply:
But if you think about it, `NaN` and floats aren't the wrong type, are they? They're still `Number`s.

They are invalid arguments because they lie outside of the rational input range.

Reply:
I didn't realize the type was only allowed to be a JS type, not a type in the broader sense (like integer).